src/services/subscriptions/subscription.hooks.js

```diff
@@ -0,0 +1,73 @@
+const errors = require('@feathersjs/errors');
+const { PROJECT_TYPE } = require('../../models/subscription.model');
+const onlyInternal = require('../../hooks/onlyInternal');
+
+const validatePayload = () => async context => {
+  const { app, data, params } = context;
+  const { user } = params;
+  if (!user) {
+    throw new errors.NotAuthenticated();
+  }
+  const { projectTypeId, projectType, enabled } = data;
+  if (!projectTypeId || !projectType || enabled === undefined) {
+    throw new errors.BadRequest('projectTypeId and projectType and enabled are required');
+  }
+
+  let service;
+  switch (projectType) {
+    case PROJECT_TYPE.MILESTONE:
+      service = app.service('milestones');
+      break;
+    case PROJECT_TYPE.CAMPAIGN:
+      service = app.service('campaigns');
+      break;
+    case PROJECT_TYPE.DAC:
+      service = app.service('dacs');
+      break;
+    default:
+      throw new errors.BadRequest('Invalid projectType');
+  }
+  const projects = await service.find({
```

Comment:
We don't need `projectId` as we discussed, so it would be better to change this query with mongodb count query to check project exists or not. Also using Model instead of Service is preferred.

Reply:
done